src/com/google/enterprise/gsafeed/FeedHelper.java

```diff
@@ -184,7 +149,34 @@ public static void marshal(Gsafeed feed, OutputStream out)
     marshaller.setProperty("com.sun.xml.internal.bind.xmlHeaders",
         "<?xml version=\"1.0\" encoding=\"UTF-8\"?>"
         + System.getProperty("line.separator")
-        + "<!DOCTYPE gsafeed PUBLIC \"-//Google//DTD GSA Feeds//EN\" \"\">");
+        + "<!DOCTYPE " + rootElement
+        + " PUBLIC \"" + PUBLIC_ID + "\" \"\">");
     marshaller.marshal(feed, out);
   }
+
+  @SuppressWarnings("unchecked")
+  private T unmarshal(InputStream inputStream, boolean validating)
```

Comment:
Maybe move this up after the callers, to make the diffs more clear (the structure is just different, it would be near the callers, rather than pushing private helpers to the bottom)?